### ISSUE TITLE
Tags index: Sort page list by title

### ISSUE THREAD

**Issue by normandy7** [squidfunk/mkdocs-material#5613]
### Description

**Idea**: For the tags index, add option to sort the listed pages alphabetically by rendered title, as opposed to nav order or file path.

### Related links

- #5612

### Use Cases

If the list of pages under each tag is not sorted alphabetically, it ends up looking rather arbitrary.

Alphabetical sorting would make it easier for the reader to process the list and find relevant pages.

### Visuals

[Example tags index page](https://docs.neptune.ai/tags/) (neptune.ai docs)

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Released as part of [9.1.21+insiders-4.39.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.39.0).

**Comment by normandy7** [comment]
Unfortunately this doesn't solve our issue, as the pages are not sorted by the same title property that is used to render the page links. (See screenshot.)

It would seem that the nav title is used for sorting, whereas the `title` meta is used to render the link text. So visually the sorting order still doesn't quite match. It would also seem that pages starting with lowercase letters come first, whereas we'd prefer case-insensitivity.

In summary, we're happy with the `title` meta being used to render the index; we'd just like the sorting to agree with it.

![image](https://github.com/squidfunk/mkdocs-material/assets/7567953/cbab7397-2def-4994-abf3-3e9be7271918)

**Comment by squidfunk** [comment]
Thanks for the feedback. Indeed, we should add `page.meta.title` to prioritze over `page.title`. Reopening.

**Comment by squidfunk** [comment]
Hmm, I'm having trouble to reproduce the problem you report. If I set a meta `title`, the pages are sorted correctly. I'm not sure what configuration you're using. Would you be able to provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) a create a new issue? It's like a quick fix then, happy to look into it right away!

**Comment by normandy7** [comment]
I'm not sure I can, as I don't have access to Insiders locally. All I can say with certainty is that the surface form of the index items corresponds to the title meta, whereas the sorting is clearly governed by something else.

I understand if you can't do much without a reproduction, but just in case, here's some of our config:

```
plugins:
  - meta
  - search:
      separator: '[\s\-,:!=\[\]()"/]+|\.(?!\d)'
  - tags:
      tags_allowed:
        - admin
        - collaboration
        - comparison
        - debugging
        - model-registry
        - platform
        - reproducing
        - versioning
        - source-code
        - time-series
        - computer-vision
        - nlp
        - tabular-data
      tags_file: tags.md
      tags_pages_compare: !!python/name:material.plugins.tags.page_title
  - typeset
```

We do have some "unorthodox" nav titles. As another example, the items in the screenshot have the following title configurations.

![image](https://github.com/squidfunk/mkdocs-material/assets/7567953/a4c7c464-9617-4581-8cd0-bca30699d80b)

1. `title: Project API reference`
    - nav: `<code>Project</code>: api/project.md`
    - h1: `Project`
2. `title: management API reference`
    - nav: `<code>management</code>: api/management.md`
    - h1: `management`
3. `title: Workspaces and projects`
    - nav: `About workspaces and projects: about/workspaces_and_projects.md`
    - h1: `Workspaces and projects`
4. `title: Adding tags`
    - nav: `usage/adding_tags.md`
    - h1: `Adding tags`

If the issue is likely to be in our config, we'll just stay put and wait for the new tags plugin, or see if we can achieve a good balance of formatting and sensible sorting through further troubleshooting.

**Comment by squidfunk** [comment]
Thanks for providing the additional info, but I'm having a hard time to figure out how to fit all of this together to observe the problem, so we can fix it. I'm sorry, but we really need a reproduction to fix it, since we can't manage to observe it 😅
